azurerm/resource_arm_firewall.go

```diff
@@ -127,6 +127,21 @@ func resourceArmFirewallCreateUpdate(d *schema.ResourceData, meta interface{}) e
 		},
 	}
 
+	exists, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(exists.Response) {
+			return fmt.Errorf("Error checking for presence of existing Firewall %q (Resource Group %q): %s", name, resourceGroup, err)
+		}
+	}
+	if exists.ID != nil && *exists.ID != "" {
+		if exists.AzureFirewallPropertiesFormat != nil {
+			props := *exists.AzureFirewallPropertiesFormat
+			parameters.AzureFirewallPropertiesFormat.ApplicationRuleCollections = props.ApplicationRuleCollections
+			parameters.AzureFirewallPropertiesFormat.NetworkRuleCollections = props.NetworkRuleCollections
+			parameters.AzureFirewallPropertiesFormat.NatRuleCollections = props.NatRuleCollections
+		}
+	}
```

Comment:
I believe we should be able to wrap this in: `!d.IsNewResource()` (so we only look this info up for Updates) to make this:

```
if !d.IsNewResource() {
  existing, err := client.Get(ctx, resourceGroup, name)
  if err != nil {
    if utils.ResponseWasNotFound(exists.Response) {
      return fmt.Errorf("Firewall %q was not found in Resource Group %q!", name, resourceGroup)
    }
    
    return fmt.Errorf("Error retrieving existing Firewall %q (Resource Group %q): %s", name, resourceGroup, err)
  }
  
  if existing.AzureFirewallPropertiesFormat == nil {
    return fmt.Errorf("Error retrieving existing Rules - `props` was nil")
  }
  
  props := *exists.AzureFirewallPropertiesFormat
  parameters.AzureFirewallPropertiesFormat.ApplicationRuleCollections = props.ApplicationRuleCollections
  parameters.AzureFirewallPropertiesFormat.NetworkRuleCollections = props.NetworkRuleCollections
  parameters.AzureFirewallPropertiesFormat.NatRuleCollections = props.NatRuleCollections
}
```

WDYT?

Reply:
Yup makes sense